NitroxClient/Communication/Packets/Processors/SeamothModuleActionProcessor.cs

```diff
@@ -0,0 +1,80 @@
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class SeamothModuleActionProcessor : ClientPacketProcessor<SeamothModulesAction>
+    {
+        private readonly IPacketSender packetSender;
+
+        public SeamothModuleActionProcessor(IPacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+        public override void Process(SeamothModulesAction packet)
+        {
+            using (packetSender.Suppress<SeamothModulesAction>())
+            using (packetSender.Suppress<ItemContainerRemove>())
+            {
+                GameObject _gameObject = GuidHelper.RequireObjectFrom(packet.Guid);
+                SeaMoth seamoth = _gameObject.GetComponent<SeaMoth>();
+                if (seamoth != null)
+                {
+                    if (packet.TechType == TechType.SeamothElectricalDefense)
+                    {
+                        float[] chargearray = (float[])seamoth.ReflectionGet("quickSlotCharge");
+                        float charge = chargearray[packet.SlotID];
+                        float slotCharge = seamoth.GetSlotCharge(packet.SlotID);
+                        GameObject gameObject = global::Utils.SpawnZeroedAt(seamoth.seamothElectricalDefensePrefab, seamoth.transform, false);
+                        ElectricalDefense component = gameObject.GetComponent<ElectricalDefense>();
+                        component.charge = charge;
+                        component.chargeScalar = slotCharge;
+                    }
+
+                    if (packet.TechType == TechType.SeamothTorpedoModule)
+                    {
+                        Transform muzzle = (packet.SlotID != seamoth.GetSlotIndex("SeamothModule1") && packet.SlotID != seamoth.GetSlotIndex("SeamothModule3")) ? seamoth.torpedoTubeRight : seamoth.torpedoTubeLeft;
+                        ItemsContainer storageInSlot = seamoth.GetStorageInSlot(packet.SlotID, TechType.SeamothTorpedoModule);
+                        TorpedoType torpedoType = null;
+                        for (int i = 0; i < seamoth.torpedoTypes.Length; i++)
+                        {
+                            if (storageInSlot.Contains(seamoth.torpedoTypes[i].techType))
+                            {
+                                torpedoType = seamoth.torpedoTypes[i];
+                                break;
+                            }
+                        }
+                        //Original Function use Player Camera need parse owner camera values
+                        TorpedoShot(storageInSlot, torpedoType, muzzle,packet.Forward,packet.Rotation);
+                        
+
+                    }
+                }
+            }
+        }
+
+        public static bool TorpedoShot(ItemsContainer container, TorpedoType torpedoType, Transform muzzle, Vector3 forward, Quaternion rotation)
+        {
+            if (torpedoType != null && container.DestroyItem(torpedoType.techType))
```

Comment:
Will DestroyItem pick a random item?  If it is random, will containers become out of sync since item guids no longer match?

Reply:
that is original code i dont found yet container of that but i tink can change sender client take ramdom and send guid for rest client